lib/bookmark_manager.rb

```diff
@@ -7,22 +7,27 @@ def initialize(user)
     @user = user
   end
 
-  def create(post_id:, name: nil, reminder_type: nil, reminder_at: nil, options: {})
-    post = Post.find_by(id: post_id)
+  def create(post_id:, topic_id: nil, name: nil, reminder_type: nil, reminder_at: nil, options: {})
+    if post_id.blank? && topic_id.present?
+      bookmark_for_topic = true
+      topic = Topic.find_by(id: topic_id)
+    else
+      post = Post.find_by(id: post_id)
+      topic = post&.topic
+    end
     reminder_type = parse_reminder_type(reminder_type)
 
-    # no bookmarking deleted posts or topics
-    raise Discourse::InvalidAccess if post.blank? || post.topic.blank?
+    raise Discourse::InvalidAccess if (!bookmark_for_topic && post.blank?) || topic.blank?
```

Comment:
I think it is better if this is pushed to the controller action because right now the error is too generic. Is `post_id` wrong or is `topic_id` wrong when the error is raised.